### PR TITLE
fixed precedence of sequence expression in computed property name

### PR DESCRIPTION
**Pull request by sanex3339**
Fixes https://github.com/javascript-obfuscator/javascript-obfuscator/issues/638
@michaelficarra this is a high priority bug for my package. Please check this.

### CODE REVIEW

File: escodegen.js [hunk 1]
```diff
@@ -983,7 +983,13 @@
...
         }
 
-        result.push(this.generateExpression(expr, Precedence.Sequence, E_TTT));
+        result.push(this.generateExpression(
+            expr,
+            expr.type === Syntax.SequenceExpression
+                ? Precedence.Primary
+                : Precedence.Sequence,
+            E_TTT
+        ));
```
Comment (michaelficarra) [inline escodegen.js:992]:
```suggestion
        result.push(this.generateExpression(expr, Precedence.Assignment, E_TTT));
```

See https://tc39.es/ecma262/#prod-ComputedPropertyName.

File: test/compare-acorn-es6/class-declaration.expected.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class ComputedKey {
+    [n1 + n2]() {
+    }
+}
+class ComputedKeyWithParenthesis {
+    [(n1 + n2) * n3]() {
```
Comment (michaelficarra) [inline test/compare-acorn-es6/class-declaration.expected.js:6]: A more helpful test would use an assignment (testing the other boundary condition) or a `yield`.

File: escodegen.js (hunk 1 above)
Comment (sanex3339) [reply escodegen.js]: Thank you, https://github.com/estools/escodegen/pull/415/commits/13e3eb826b50ebcb5dd10efd9515f22bf9f73c3a

File: test/compare-acorn-es6/class-declaration.expected.js (hunk 2 above)
Comment (sanex3339) [reply test/compare-acorn-es6/class-declaration.expected.js:6]: Not sure what did you meant.

Comment (sanex3339) [reply test/compare-acorn-es6/class-declaration.expected.js]: Did you meant this?
189e8c5

File: test/compare-acorn-es6/class-declaration.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class ComputedKey {
+    [n1 + n2]() {}
+}
+class ComputedKeyWithParenthesis {
+    [n1 = (n2 + n3) * n4]() {}
```
Comment (michaelficarra) [inline test/compare-acorn-es6/class-declaration.js:5]:
```suggestion
class AssignmentExpressionAsKey {
    [n1 = n2]() {}
```

Comment (sanex3339) [reply test/compare-acorn-es6/class-declaration.js]: https://github.com/estools/escodegen/pull/415/commits/25eb2b8f14ad4bf7c06030751f5b8f1d13dce0eb 
Thank you

**Comment by sanex3339** [comment]
Thank you
